src/Middleware/SubstituteBindings.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace LaravelDoctrine\ORM\Middleware;
+
+use Closure;
+use Doctrine\ORM\EntityNotFoundException;
+use Illuminate\Contracts\Routing\Registrar;
+use Illuminate\Routing\Route;
+use LaravelDoctrine\ORM\IlluminateRegistry;
+use ReflectionFunction;
+use ReflectionMethod;
+
+class SubstituteBindings
+{
+    /**
+     * The router instance.
+     *
+     * @var \Illuminate\Contracts\Routing\Registrar
+     */
+    protected $router;
+
+    /**
+     * @var IlluminateRegistry
+     */
+    protected $registry;
+
+    /**
+     * Create a new bindings substitutor.
+     *
+     * @param Registrar          $router
+     * @param IlluminateRegistry $registry
+     */
+    public function __construct(Registrar $router, IlluminateRegistry $registry)
+    {
+        $this->router   = $router;
+        $this->registry = $registry;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  \Closure                 $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        $this->substituteImplicitBindings($request->route());
+
```

Comment:
I think the call to `$this->router->substituteBindings($request->route())` could have stayed.